mizar/common/common_fornax.py

```diff
@@ -0,0 +1,75 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2021 The Authors.
+
+# Authors: Jun Shao <@jshaofuturewei
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:The above copyright
+# notice and this permission notice shall be included in all copies or
+# substantial portions of the Software.THE SOFTWARE IS PROVIDED "AS IS",
+# WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+# TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+# FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR
+# THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import logging
+from mizar.common.common import *
+
+logger = logging.getLogger()
+
+def get_cluster_gateway_host_ip(core_api):
+    # Read gateway_host_ip from configmap
+    cluster_gateway_config = kube_read_config_map(core_api,  "cluster-gateway-config", "default")
+    gateway_host_ip = ""
+    if cluster_gateway_config:
+        gateway_host_ip = cluster_gateway_config.data["gateway_host_ip"]
+        logger.info("The gateway host ip is {}".format(gateway_host_ip))
+    else:
+        logger.info("No gateway host is configured.")
+    return gateway_host_ip
+
+def get_cluster_gateway_droplet(droplets, cluster_gateway_host_ip):
+    cluster_gateway_droplet = ""
+    for dd in droplets:
+        if dd.ip == cluster_gateway_host_ip:
+            cluster_gateway_droplet = dd
+            logger.info("A droplet {} has been selected as cluster gateway.".format(dd.ip))
```

Comment:
Did you mean "Droplet {} ... " instead of "A droplet {} ..." ?

Reply:
Updated